main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -572,7 +572,21 @@ object Safety {
     //
     val err2 = checkHeadPredicate(c0.head, unsafeLatVars)
 
-    err1 concat err2
+    val err3 = c0.body.flatMap(s => checkBodyPattern(s))
+
+    err1 concat err2 concat err3
+  }
+
+  // Performs safety check on the pattern of an atom body.
+  private def checkBodyPattern(p0: Predicate.Body): List[CompilationMessage] = p0 match {
+    case Predicate.Body.Atom(_,_,_,_,terms,_,loc) =>
+      terms.foldLeft[List[SafetyError]](Nil)((acc, term) => term match {
+        case Pattern.Var(_, _, _)
```

Comment:
(As a style choice) we don't use Or patterns, instead preferring to write each case. But of course you can still have your wildcard case.